src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/logs-highlightsearch.pipe.ts

```diff
@@ -0,0 +1,26 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'highlightSearch'
+})
+export class LogsHighlightSearchPipe implements PipeTransform {
+  transform(value: string, args: string): any {
+    if (!args) {
+      return value;
+    }
+    args = this.escapeRegExp(args);
+    const regex = new RegExp(args, 'gi');
+    const match = value.match(regex);
+
+    if (!match) {
+      return value;
+    }
+
+    return value.replace(regex, '<mark>$&</mark>');
+  }
+
+  private escapeRegExp(str: string) {
+    // $& means the whole matched string
+    return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
+  }
+}
```

Comment:
Maybe it makes more sense if you declare the variable with some meaningful names so that we'll understand its purpose just by looking at its name ;)